frontend/lib/src/components/elements/CodeBlock/StreamlitSyntaxHighlighter.tsx

```diff
@@ -48,6 +53,39 @@ export default function StreamlitSyntaxHighlighter({
           style={{}}
           lineNumberStyle={{}}
           showLineNumbers={showLineNumbers}
+          wrapLongLines={wrapLines}
+          // Fix bug with wrapLongLines+showLineNumbers (see link below) by
+          // using a renderer that wraps individual lines of code in their
+          // own spans.
+          // https://github.com/react-syntax-highlighter/react-syntax-highlighter/issues/376
+          renderer={({ rows, stylesheet, useInlineStyles }) =>
+            rows.map((row, index) => {
```

Comment:
nit: as a small optimization, we could define the code below as part of a `useCallback` and only provide it to the `renderer` if `showLineNumbers && wrapLines` are both true and otherwise pass in undefined (since the fix is only required in this specific as far as I understood it)

Reply:
Great point. Done.